src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
```

Comment:
Well, `md_config_t::diff_setting(const md_config_t *other, map<string, pair<string, string>> *diff, 
const string& ceph_setting, bool show_unchanged)`  and `void md_config_t::diff(const md_config_t *other, map<string,pair<string,string> > *diff, set<string> *unknown)` quite different signatures, so I wonder whether or not the trade offs on consolidating them would make maintenance troublesome.